### PR TITLE
Add transform_to method to CoordSys3D

### PR DESCRIPTION
**Pull request by szymag**
This PR adds a method `transform_to` which is similar to method that works for translation and rotation. 
I also add to documentation short description about that.

This PR also fixes `transformation_from_parent` method. `Coordsys3D` doesn't have attribute `parent but has `_parent`.

### CODE REVIEW

**Comment by szymag** [comment]
ping @Upabjojr

**Comment by Upabjojr** [comment]
This looks OK.

**Comment by Upabjojr** [comment]
I was thinking about the `transform_to` function name. Are we sure it's a good name?

**Comment by Upabjojr** [comment]
The name `transform_to` makes me think about a way to transform something into a new coordinate system, not about the creation of a new coordinate system. Maybe we should think about a better name.

**Comment by szymag** [comment]
I'm open for discussion.
What about `transform_new`, like `locate_new`?

**Comment by Upabjojr** [comment]
I think that the word _transform_ sounds a bit like something that changes. We need to make sure that it's a new coordinate system that is being created. What about `create_new`?

**Comment by szymag** [comment]
I like `create_new` but now it's too general name IMO. Here we can set only `transformation`. I'd say that in that case we should be able to set every possible parameters

**Comment by Upabjojr** [comment]
Well, isn't the transformation the same as every possible parameter?

**Comment by szymag** [comment]
Not at that time. We have also `rotation_matrix` and `location`. 
If we remove them in the future and leave only `transformation`, `create_new` will OK

File: sympy/vector/coordsysrect.py [hunk 1]
```diff
@@ -946,6 +950,40 @@ def orient_new_quaternion(self, name, q0, q1, q2, q3, location=None,
...
+            Used for simple str printing.
+
+        Examples
+        ========
+
+        >>> from sympy.vector import CoordSys3D
+        >>> a = CoordSys3D('a')
+        >>> b = a.transform_to('b', transformation='spherical')
+        >>> b.transformation_to_parent()
+        (b.x*sin(b.y)*cos(b.z), b.x*sin(b.y)*sin(b.z), b.x*cos(b.y))
```
Comment (Upabjojr) [inline sympy/vector/coordsysrect.py]: these would be clearer if you use `r`, `theta`, `phi`.

Comment (szymag) [inline sympy/vector/coordsysrect.py]: OK

**Comment by Upabjojr** [comment]
> Not at that time. We have also rotation_matrix and location.
> If we remove them in the future and leave only transformation, create_new will OK

Well, everything is eventually stored into `transformation` eventually. Did you have a look at my code refactory?

I believe `create_new` is a good name candidate. Do you have any other suggestion?

**Comment by szymag** [comment]
> . Did you have a look at my code refactory?

Yes

I'm talking from user perspective. `create_new` means that I should be able to create any desired coordinate system. Now I can't set `rotation_matrix` if I want. To do that I need to use another methods.

**Comment by szymag** [comment]
I believe that we should reflect in the name that we're working with transformation equations.

**Comment by Upabjojr** [comment]
> Now I can't set rotation_matrix if I want. To do that I need to use another methods.

Yes, you can.

**Comment by Upabjojr** [comment]
You can set `transformation=(rotation_matrix, [0, 0, 0])`

**Comment by szymag** [comment]
But It's not the clearest way to do that, especially for user who worked with `vector` package for longer time. As I said, this name will be OK when `transformation` will only one way for creating transformation.

**Comment by Upabjojr** [comment]
what name do you suggest?

**Comment by szymag** [comment]
I suggested but you didn't like it:)

**Comment by Upabjojr** [comment]
I am not sure about using the word _transform_.

**Comment by szymag** [comment]
I understand your concern. 
Would you like to keep in constructor only `transformation` and remove in the future `rotation_matrix` and `location`? 
If so, I think we can use `create_new`

**Comment by Upabjojr** [comment]
we could add a parser for `transformation=matrix` and `transformation=3 element list` for the translation.

**Pull request merged**